v8context.h

```diff
@@ -31,12 +31,17 @@ class V8Context {
 
   void Throw(const char* errmsg);
 
+  bool HasTerminated() const;
+
  private:
   std::string report_exception(v8::TryCatch& try_catch);
 
   v8::Isolate* mIsolate;
   v8::Persistent<v8::Context> mContext;
   std::string mLastError;
+
+  // If true, the last JS execution was terminated prematurely
+  bool mTerminated;
```

Comment:
IMO worth updating the V8Context::V8Context constructor (v8context.cc:111) with an explicit mTerminated(false) initializer. It's true that you must always reinitialize to false on entry to eval or apply anyway, but it seems more sensible for the state to be predictably false before eval or apply has ever been called.

(I'm annoyed that GitHub PRs don't allow me to put this comment up there...)